### PR TITLE
Do not throw on "empty" package; emit warning instead

### LINKED ISSUE #2327: Graceful handling of asset-only packages.
When a package has no .dart files in it, `dartdoc` fails with the following output:
````
Unhandled exception:
dartdoc could not find any libraries to document
#0      Dartdoc.generateDocs (package:dartdoc/dartdoc.dart:218:9)
<asynchronous suspension>
#1      main (file:///project/pkg/pub_dartdoc/bin/pub_dartdoc.dart:27:33)
<asynchronous suspension>
#2      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#3      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
````

Instead, `dartdoc` should emit a warning, and still generate the `index.html` file based on the `README.md`.

Example package from pub.dev: `cupertino_icons`, `bulma_min`.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #2327.

It's valid to have a dart package without libraries, so only emit a (silence-able) warning and generate the index and other bits as normal if this happens.

example output from the new version:
```
jcollins-macbookpro2:dartdoc jcollins$ PACKAGE_NAME=bulma_min grind build-pub-package
grinder running build-pub-package

build-pub-package
build-bulma_min: + PUB_CACHE=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA PATH=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA/bin:/Users/jcollins/mdproxy/bin:/Users/jcollins/google-cloud-sdk/bin:/Users/jcollins/bin:/usr/local/bin:/Users/jcollins/.pub-cache/bin:/Users/jcollins/dart/all_sdks/current/bin:/Users/jcollins/dart/flutter/bin:/Users/jcollins/depot_tools:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin pub cache add bulma_min
build-bulma_min: Downloading bulma_min 0.7.4...
  copying /var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA/hosted/pub.dartlang.org/bulma_min-0.7.4 to /var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/bulma_minvc8AJM
build-bulma_min: + (cd "/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/bulma_minvc8AJM" && PUB_CACHE=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA PATH=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA/bin:/Users/jcollins/mdproxy/bin:/Users/jcollins/google-cloud-sdk/bin:/Users/jcollins/bin:/usr/local/bin:/Users/jcollins/.pub-cache/bin:/Users/jcollins/dart/all_sdks/current/bin:/Users/jcollins/dart/flutter/bin:/Users/jcollins/depot_tools:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin pub get)
build-bulma_min: Resolving dependencies...
build-bulma_min: Got dependencies!
build-bulma_min: + (cd "/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/bulma_minvc8AJM" && PUB_CACHE=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA PATH=/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/pubcachet05MCA/bin:/Users/jcollins/mdproxy/bin:/Users/jcollins/google-cloud-sdk/bin:/Users/jcollins/bin:/usr/local/bin:/Users/jcollins/.pub-cache/bin:/Users/jcollins/dart/all_sdks/current/bin:/Users/jcollins/dart/flutter/bin:/Users/jcollins/depot_tools:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin /Users/jcollins/dart/all_sdks/2.10.0-156.0.dev/bin/dart --enable-asserts /Users/jcollins/dart/dartdoc/bin/dartdoc.dart --json --link-to-remote --show-progress)
build-bulma_min: Documenting bulma_min...
build-bulma_min:   warning: package:bulma_min has no documentable libraries, from package-bulma_min: file:///private/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/bulma_minvc8AJM
build-bulma_min:             Dartdoc could not find any public libraries to document in package:bulma_min,
build-bulma_min:             but documentation was requested.  This might be expected for an
build-bulma_min:             asset only package, in which case, disable this warning in your
build-bulma_min:             dartdoc_options.yaml file.
build-bulma_min: Initialized dartdoc with 38 libraries in 15.8 seconds
build-bulma_min: Validating docs...
build-bulma_min: found 1 warning and 0 errors
build-bulma_min: Documented 0 public libraries in 0.7 seconds
build-bulma_min: dartdoc could not find any libraries to document
build-bulma_min: Success! Docs generated into /private/var/folders/b8/jg3_spkj02d51bcz3fsb5jjc000b33/T/bulma_minvc8AJM/doc/api

finished in 23.2 seconds
```

### CODE REVIEW

File: lib/src/model/package_graph.dart [hunk 1]
```diff
@@ -72,7 +72,7 @@ class PackageGraph {
     await Future.wait(precacheLocalDocs());
     _localDocumentationBuilt = true;
 
-    // Scan all model elements to insure that interceptor and other special
+    // Traverse all can all model elements to insure that interceptor and other special
```
Comment (srawlins) [inline lib/src/model/package_graph.dart:75]: I can't parse this sentence... typo?

**Review by srawlins — APPROVED** [review]

File: lib/src/model/package_graph.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/model/package_graph.dart]: yes.  This was a draft PR, I didn't realize you were going to jump on it right away.  Hadn't done my final typo scan yet :-)

**Comment by jcollins-g** [comment]
Ready, now.  PTAL?

**Review by srawlins — APPROVED** [review]
Love it!

**Pull request merged**